### DISCUSSION TITLE
Date validation

### DISCUSSION THREAD

**Discussion by renrom** [Budibase/budibase#11078] [Help] [answered]
I have a date field where it is not allowed to enter a date in the future. Can i somehow validate that in the validation?

**Comment by ConorWebb96** [comment, marked as answer]
Hey @renrom,

Yes this should be possible, you will have to use the configure validation button on the date picker. I've created some steps as well as a JS function that may help you in this case below;

1. Add your form and date components
2. Click on your date component.
3. On the right side of the screen there should be a button alongwith the date component information at the bottom called `Configure validation` click this.
4. Change the settings within similar to what i've shown below
<img width="100%" alt="Screenshot 2023-06-30 at 08 08 43" src="https://github.com/Budibase/budibase/assets/126772285/63c6a6c2-6ef8-4a67-9252-48651a65581d">
5. Add this JS code to bindings area were it says JS function in the above screenshot.
<img width="100%" alt="Screenshot 2023-06-30 at 08 09 23" src="https://github.com/Budibase/budibase/assets/126772285/c5910d85-4f66-4d11-b2ac-210c436e91c5">
```
const newDate = new Date();
const formattedDate = newDate.toISOString();
return formattedDate;
```

**Demo below of this working**
![date validation](https://github.com/Budibase/budibase/assets/126772285/be4aa9a7-66d1-43b6-affa-3486a0662be4)

A small note it will factor in time so, as the above code is setting the time to 00.00, you can alter this function to be 23.59 on that date instead and this won't matter.

I hope this helps.

**Reply by renrom** [reply]
Great! Thanks for your quick reply and the very clear example.
